README.md

```diff
@@ -1512,6 +1512,53 @@ would enable the use of a `@fold` on the `adjacent_animal` vertex field of `Foo`
 }
 ```
 
+## Schema Generation
+To generate a GraphQL schema, we first map a database schema into an intermediate representation:
+the `SchemaGraph`.
+
+#### SchemaGraph
+* `SchemaGraph`
+    * Keeps track of a schema's transitive inheritance structure.
+    * Keeps track of a schema's indexes. Indexes are not used in the GraphQL schema generation and 
+      have a provisional API.
+    * Represents a database's schema through `SchemaElement` objects.
+        * `SchemaElement`
+            * Can be one of three kinds:
+                * `VertexType`, which represent schema vertices.
+                * `EdgeType`, which represent relations between vertices.
+                * `NonGraphElement`, which represents a schema element with usually no connections
+                   to other classes. A `NonGraphElement` may only have connections if it's abstract 
+                   and all its non-abstract subclasses are `VertexTypes`. 
+            * Have the following attributes:
+                * `name`, indicating the name of the class.
+                * `abstract`, indicating whether the class is abstract or not.
+                * `properties`, which are represented using GraphQLTypes.
+                * `in_connections`/`out_connections`, which represent the set of possible adjacent 
+                  `SchemaElements`.
+                    * For an `EdgeType`, its connections represent the set of allowed `VertexTypes` 
+                      and `NonGraphElements` at each end of the edge.
+                    * For `VertexType` and `NonGraphElement` objects, their connections represent
+                      the set of edges connected to them.
+            * `EdgeType` objects also have:
+                * `base_in_connection`/`base_out_connection`, which are the classes allowed at each 
+                  edge end that are the superclass of all other classes allowed at the edge end.
+
+#### Mapping an OrientDB schema to the SchemaGraph. 
+* We map classes that inherit from the base OrientDB vertex and edge classes to `VertexTypes` and 
+  `EdgeTypes` respectively. 
+* We map the remaining set of classes to `NonGraphElements`.
+
+#### Mapping the SchemaGraph to the GraphQLSchema
+* We map non-abstract `VertexTypes` to `GraphQLObjects`.
```

Comment:
Try to avoid pluralizing type names, and instead rework the sentence structure so that the plural goes elsewhere and the type name remains something you can copy-paste into your editor's search bar: `VertexType`, `GraphQLObject` etc.

Also, I don't think `GraphQLObject` exists, I think it was maybe named `GraphQLObjectType` or something like that. I think this is a problem that appears in other places and for other type names as well.